documentation/Query-Syntax.md

```diff
@@ -119,7 +119,36 @@ Compares an input string with a regular expression string and returns whether it
 
 ## objectProperties
 
-Used to extract values from current state, for example, from `user`, `organisation` or `form` (i.e. previously entered responses) data. Any object to be evaluated must be passed in to the `evaluateExpression` function in its parameters object, with the field name identical to the object name (see below).
+For extracting values from local state objects (e.g. `user`, `organisation` or `form`) as part of queries. This might be needed if a question's visibility condition relies on the answer to a previous question, for example. All objects to be queried are to be passed in in an array in the field `objects`. See **Usage** below for detailed overview of arguments.
+
+- Input: A single child node whose `value` is an object with the following properties:
+  - `objectIndex` (optional) -- the index of the `objects` array that contains the object this node is interested in. If this property is not defined, it defaults to `0`, so you'd usually only include it if you were supplying more than one object to the whole expression.
+  - `property` -- the name of the field whose value is to be extracted. Can be a nested property, written in dot notation (e.g. `questions.q2`) (but cannot yet )
+- Output: the value specified in `property`, so could be any type.
```

Comment:
```suggestion
 - Input (i.e. children `array`):
_Note: Each node can only be dealing with one object property!_ 
The array `objects` is passed for more than one evaluationExpression object and each node will always associated to a single property only.
    - 1st child node returns an object with the following properties:
      - `objectIndex` (optional) -- index of one object in the `objects` array of the object this node is interested in. Defaults to `0` -- only include it if you were supplying more than one object to the whole expression.
      - `property` -- the name of the field whose value is to be extracted. Can be a nested property, written in dot notation (e.g. `questions.q2`) **(but cannot yet )**
- Output:
  - Value specified in `property` of any type.
```

Reply:
Okay, have reworded it and tried to include your added information. The (but cannot yet) was an incomplete sentence 😊 -- have fixed to "(but cannot yet traverse into arrays.)"